src/test/java/com/iota/iri/network/TransactionRequesterTest.java

```diff
@@ -76,6 +79,48 @@ public void instance() throws Exception {
 
     }
 
+    @Test
+    public void popEldestTransactionToRequest() throws Exception {
+        TransactionRequester txReq = new TransactionRequester(tangle, snapshotProvider, mq);
+        // Add some Txs to the pool and see if the method pops the eldest one
+        Hash eldest = TransactionViewModelTest.getRandomTransactionHash();
+        txReq.requestTransaction(eldest, false);
+        txReq.requestTransaction(TransactionViewModelTest.getRandomTransactionHash(), false);
+        txReq.requestTransaction(TransactionViewModelTest.getRandomTransactionHash(), false);
+        txReq.requestTransaction(TransactionViewModelTest.getRandomTransactionHash(), false);
```

Comment:
Just heads up

Technically it is not a good idea to use randomness in unit tests (you want them to be consistent)
Unfortunately, the current codebase encourages you to write this way.

For this specific unit test it is less of a problem and I don't want to currently fix this problem which is not related to you or the task at hand.

Just FYI

Reply:
I thought the same when I was coding but well...